### PR TITLE
Restructure the REMEMBERED_URLS_KEY dictionary to include the expirat…

### LINKED ISSUE #36: FANCY_REMEMBER_ALL_URLS option and performance over time
Are there known limitations to this option? I had enabled it to give more visibility to our system and at about ~22,000 entries, the endpoints that were being tracked stopped being functional. Clearing the `fancy-urls` key solved the issue immediately.

```
In [1]: v = cache.get('fancy-urls')

In [2]: len(v.keys())
Out[2]: 22717
```

Above was the final count before deleting the key.

### PR DESCRIPTION
**Pull request by YPCrumble**
…ion time of a given URL.

This will prevent the dictionary from growing over time with stale entries.
Fixes #36.

Based on comments from @peterbe in #59

### CODE REVIEW

**Comment by YPCrumble** [comment]
@peterbe this runs with your idea of a remembered URLs dict like `{ url: (cache_key, expiration_time)}`. Basically I'm storing the expiration time of each URL so that I can clean it up using the `filter_remembered_urls` utility function. 

The size of the dict can still increase infinitely, but this PR will prevent stale entries from being part of the dict which will solve the issue I was encountering (I think!).

**Comment by peterbe** [comment]
Now that the tox.ini has been updated in `main`. Thank you! ...can you update *this* branch?

**Review by peterbe — APPROVED** [review]
I only read the code. I didn't try it. Have you played with it in a local instance to see that it actually works and keeps the massive dict in check?

I'll merge it once the tests pass after checking in the latest `master`.

**Comment by peterbe** [comment]
Why is `pypy` so flaky? It happened on a `master` run too. But then it worked. 
I don't even understand the error.

**Comment by YPCrumble** [comment]
Agreed - I don't know either. I googled and couldn't find anything; if you look into the test there's also an SQLite version mismatch error that I'm not sure if it's related.

I would lean toward removing PyPy support personally and reinstating it if someone in the future wants to figure it out.

I'm not in a rush; I can test this branch on my application on staging and then master and fix any bugs I encounter before we merge the PR.

**Comment by peterbe** [comment]
> I would lean toward removing PyPy support personally and reinstating it if someone in the future wants to figure it out.

I would very much welcome if you make a PR that **comments it out** from `tox.ini` (and wherever else appropriate). NExt to the commenting out should be a mention of an issue on this repo here that explains that "Pypy tests are flaky so we commented them out. Chances are this project works in Pypy, but we have a maintainer for Pypy is simply not included int he test automation."

What do think?

File: fancy_cache/middleware.py
Comment (peterbe) [inline fancy_cache/middleware.py:113]: don't forget :)

Perhaps there should be a test in Actions that searches for any of these in the `.py` files.

**Review by peterbe — COMMENTED** [review]
I see some test failures but I honestly didn't deeply review it at this point. 
What's the status do you think?

**Comment by YPCrumble** [comment]
@peterbe thanks for catching that! I will add a checker for the debugger - I have a little static analysis suite on my site that I've been meaning to turn into a github action that I'll do soon.

I just made a few bugfixes to this branch and continuing to test on my site - I think this covers the most recent issues but should be ready to merge next week once I've tested this latest commit.

#68 and #70 are ready to merge - once all of these are ready I think we can get the release out. Thanks again for your help!

**Comment by peterbe** [comment]
Merge conflict now. But next time, the pypy tests won't stand in the way. 

Regarding the `import pdb` I think you can write a simple script that runs `git grep 'import pdb'` or something. And have it fail the test if something is found.

**Comment by YPCrumble** [comment]
@peterbe this is ready to go! Just resolved the merge conflicts. I also marked #65 as ready for review! When you're able I'd love to get the release out. Thanks again for your help with all of these changes!

**Review by peterbe — APPROVED** [review]
I'll be honest, I skimmed it. 
I know you're implementing a good pattern here and you have tests and you've been testing this locally. CI is happy and I'm happy.

**Pull request merged**
